src/PhpSpec/ObjectBehavior.php

```diff
@@ -31,9 +31,90 @@
  * @method void beConstructedWith(...$arguments)
  * @method void beConstructedThrough($factoryMethod, array $constructorArguments = array())
  * @method void beAnInstanceOf($class)
+ *
  * @method void shouldHaveType($type)
+ * @method void shouldNotHaveType($type)
+ * @method void shouldReturnAnInstanceOf($type)
```

Comment:
I see two options how this could work:

1. Through `@method` tags for methods of the described class in the particular specification

```php
class Foo
{
    public function bar()
    {
        return 'foobar';
    }
}

/**
 * @method \PhpSpec\Wrapper\Subject bar()
 */
class FooSpec extends ObjectBehaviour
{
    function it_should_bar()
    {
        $this->bar()->shouldBe('foobar');
    }
}
```

2. Through an IDE plugin that already knows the method delegations of the described class and knows that those methods return `Subject` (I already have such a plugin for PhpStorm that supports this). So if we had the matcher magic methods in the `Subject` class then, autocompletion could work for most cases. Get what I mean?